### PR TITLE
[Feature] Event Trigger Logic

### PR DESCRIPTION
**Pull request by MuhammedIrfan**
# Pull Request Template

## Description
* Added Event Trigger Logic

Please select the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] Pull Request follow only a single responsibility.
- [x] Code follows the style guidelines of this project
- [x] Performed a self-review of my own code
- [x] Commented my code, particularly in hard-to-understand areas
- [x] Made corresponding changes to the documentation
- [x] Changes generate no new warnings
- [x] Added tests that prove fix is effective or the feature works
- [x] New and existing unit tests pass locally with the changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Checked the code and corrected any misspellings

### CODE REVIEW

File: src/modules/event.module.ts [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+import { EventService } from '@services/index';
+import openwhisk from 'openwhisk';
 
 /**
  * Event Module initializes event controller
  */
 
 export const EventModule: MicrobootstrapLoader = (frameworkSettings: MicrobootstrapSettings | undefined) => {
     if (frameworkSettings) {
+        const { openwhiskApiHost, openwhiskApiKey, openwhiskNamespace } = util.loadFileConfigs();
```
Comment (HanumantappaBudihal) [inline src/modules/event.module.ts:14]: change method name to 
```suggestion
        const { openwhiskApiHost, openwhiskApiKey, openwhiskNamespace } = util.loadFileConfigurations();
```

**Review by HanumantappaBudihal — CHANGES_REQUESTED** [review]
Please go through the comments.

I notified something about the mock data preparation in test methods. Can we make more dynamic way to handle the mock data for unit testing purpose ? It's good have a some kind of mock data generator or mock data handler as separate module.

Line number from  77 to 60 in event.test.ts

" const mockEvents = [
            { "section": "system", "method": "ExtrinsicSuccess" },	            { "section": "system", "method": "ExtrinsicSuccess" },
            { "section": "system", "method": "ExtrinsicFailed" },	            { "section": "system", "method": "ExtrinsicFailed" },
            { "section": "balance", "method": "ExtrinsicFailed" },	            { "section": "balance", "method": "ExtrinsicFailed" },
        ];	        ];"

File: src/modules/event.module.ts (hunk 1 above)
Comment (MuhammedIrfan) [reply src/modules/event.module.ts:14]: loadFileConfigs is a method from external library

**Review by HanumantappaBudihal — APPROVED** [review]
Looks good

File: src/services/event.service.ts [hunk 2]
```diff
@@ -34,4 +46,17 @@ export class EventService {
             return true;
         })
     }
+
+    public triggerEventManager({ brokers, topic, event, trigger }): Observable<{
```
[the remark this answers is not in the document]
Comment (MuhammedIrfan) [reply src/services/event.service.ts:50]: It is triggerEventManager

**Review by HanumantappaBudihal — APPROVED** [review]
Looks good

**Pull request merged**